src/components/organisms/Browser.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+  <div id="browser">
+    <SearchInput placeholder="Znajdź swój przedmiot..."/>
+    <SortDropdown/>
+    <ButtonSecondary button-text="Szukaj" @click="onButtonClick"/>
+  </div>
+</template>
+
+<script>
+import SearchInput from "@/components/atoms/SearchInput.vue";
+import SortDropdown from "@/components/atoms/SortDropdown.vue";
+import ButtonSecondary from "@/components/atoms/ButtonSecondary.vue";
+
+export default {
+  // eslint-disable-next-line vue/multi-word-component-names
+  name: "Browser",
+  data() {
+    return {
+      input: '',
+      sorting: 'Najbliższe',
+    }
+  },
+  methods: {
+    onButtonClick() {
+      console.log('Search button clicked');
+      // TODO: add implementation
+    }
+  },
+  mounted() {
+    this.emitter.on('search', (data) =>
+    {
+        this.input = data.input;
+    });
+
+    this.emitter.on('sortChange', (data) =>
+    {
+      this.sorting = data.option;
```

Comment:
Change data.option to data here - 
data is being received as the name of the change itself here, i.e data.option is undefined.
(although above in data.input it's working fine)